framework/android/kodein-di-framework-android-x-viewmodel-savedstate/src/main/java/org/kodein/di/android/x/viewmodel/DIAware.kt

```diff
@@ -0,0 +1,86 @@
+@file:Suppress("UNCHECKED_CAST")
+
+package org.kodein.di.android.x.viewmodel
+
+import androidx.annotation.MainThread
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.createViewModelLazy
+import androidx.lifecycle.*
+import org.kodein.di.DIAware
+import org.kodein.di.direct
+import org.kodein.type.generic
+
+
+/**
+ * Returns a [Lazy] property delegate to access the [AppCompatActivity]'s [ViewModel], which will be obtained from a Kodein DI
+ * factory with a [SavedStateHandle] argument.
+ *
+ * ```
+ * class MyComponentActivity : AppCompatActivity(), DIAware {
+ *     override val di: DI by closestDI()
+ *     private val viewmodel: MyViewModel by viewModelSavedStateFactory()
+ * }
+ * ```
+ * @param A Extension function applies to [AppCompatActivity] which implements [DIAware]
+ * @param VM The type of [ViewModel] to obtain
+ * @param tag The bound Kodein DI tag
+ * @throws IllegalArgumentException if accessed before the Activity is attached to the Application
+ * @throws DI.NotFoundException If no factory was found.
+ * @throws DI.DependencyLoopException When calling the factory, if the value construction triggered a dependency loop.
+ */
+@MainThread
+inline fun <A, reified VM> A.viewModelSavedStateFactory(
```

Comment:
Can you rename this as `viewModelSavedState` ? 
As the factory is an internal thing et it just returns a lazy viewModel.